Kernel/Process.h

```diff
@@ -248,11 +253,11 @@ class Process final
 
     // Breakable iteration functions
     template<IteratorFunction<Process&> Callback>
-    static void for_each(Callback);
-    template<IteratorFunction<Process&> Callback>
-    static void for_each_in_pgrp(ProcessGroupID, Callback);
-    template<IteratorFunction<Process&> Callback>
-    void for_each_child(Callback);
+    static void for_each_ignoring_jails(Callback);
+
+    ErrorOr<void> for_each_with_jail_respect(Function<ErrorOr<void>(Process&)>);
+    ErrorOr<void> for_each_in_pgrp_with_jail_respect(ProcessGroupID, Function<ErrorOr<void>(Process&)>);
+    ErrorOr<void> for_each_child_with_jail_respect(Function<ErrorOr<void>(Process&)>);
```

Comment:
I'm not sure I'm sold on the `with_jail_respect` suffix here from a grammar point of view. maybe `for_each_*_respecting_jails` ? `jail_aware_for_each` ? `for_each_*_in_same_jail` ?